File: tests/fixtures/c/complex.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

struct Point {
    int x;
    int y;
};

enum Color {
    RED,
    GREEN,
    BLUE
};

typedef struct Point Point;

int add(int a, int b) {
    return a + b;
}

void process(const char* data, size_t len) {
    for (size_t i = 0; i < len; i++) {
        printf("%c", data[i]);
    }
}

int main() {
    Point p = {10, 20};
    int result = add(p.x, p.y);
    return 0;
}

File: tests/fixtures/csharp/complex.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace MyApp
{
    public class Calculator
    {
        private int _value;

        public Calculator()
        {
            _value = 0;
        }

        public int Add(int a, int b)
        {
            _value = a + b;
            return _value;
        }

        public void Print(string message)
        {
            Console.WriteLine(message);
        }
    }

    public struct Point
    {
        public double X;
        public double Y;

        public Point(double x, double y)
        {
            X = x;
            Y = y;
        }
    }

    public interface IProcessor
    {
        void Process(string data);
    }

    public enum Color
    {
        Red,
        Green,
        Blue
    }

    public static class Extensions
    {
        public static int Square(this int n)
        {
            return n * n;
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            var calc = new Calculator();
            int result = calc.Add(5, 10);
            Console.WriteLine(result);
        }
    }
}
